pandas/_libs/groupby.pyx

```diff
@@ -382,5 +382,61 @@ def group_any_all(uint8_t[:] out,
                 out[lab] = flag_val
 
 
+ctypedef fused floating:
+    float32_t
+    float64_t
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def group_add_floating(floating[:, :] out,
+                       int64_t[:] counts,
+                       floating[:, :] values,
+                       const int64_t[:] labels,
+                       Py_ssize_t min_count=0):
+    """
+    Only aggregates on axis=0
+    """
+    cdef:
+        Py_ssize_t i, j, N, K, lab, ncounts = len(counts)
+        floating val, count
+        ndarray[floating, ndim=2] sumx, nobs
+
+    if not len(values) == len(labels):
+        raise AssertionError("len(index) != len(labels)")
+
+    nobs = np.zeros_like(out)
+    sumx = np.zeros_like(out)
+
+    N, K = (<object>values).shape
+
+    with nogil:
+
+        for i in range(N):
+            lab = labels[i]
+            if lab < 0:
+                continue
+
+            counts[lab] += 1
+            for j in range(K):
+                val = values[i, j]
+
+                # not nan
+                if val == val:
+                    nobs[lab, j] += 1
+                    sumx[lab, j] += val
+
+        for i in range(ncounts):
+            for j in range(K):
+                if nobs[i, j] < min_count:
+                    out[i, j] = NAN
+                else:
+                    out[i, j] = sumx[i, j]
+
+
+group_add_float32 = group_add_floating
+group_add_float64 = group_add_floating
```

Comment:
group_add_float32 = group_add_floating[float32_t]
group_add_float64 = group_add_floating[float64_t]

or even better, track down the places where these are used and call the non-specialized one directly.

Probably also shorten the name to just group_add